tests/bsim/bluetooth/host/att/long_read/testlib/att_write.c

```diff
@@ -0,0 +1,74 @@
+/* Copyright (c) 2023 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/gatt.h>
+#include <zephyr/bluetooth/att.h>
+#include <zephyr/bluetooth/l2cap.h>
+#include <zephyr/bluetooth/uuid.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/__assert.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(att_write, LOG_LEVEL_DBG);
+
+struct bt_testlib_att_write_closure {
+	uint8_t att_err;
+	struct bt_gatt_write_params params;
+	struct k_mutex lock;
+	struct k_condvar done;
+};
+
+static void att_write_cb(struct bt_conn *conn, uint8_t att_err, struct bt_gatt_write_params *params)
+{
+	struct bt_testlib_att_write_closure *ctx =
+		CONTAINER_OF(params, struct bt_testlib_att_write_closure, params);
+
+	k_mutex_lock(&ctx->lock, K_FOREVER);
+
+	ctx->att_err = att_err;
+
+	k_condvar_signal(&ctx->done);
+	k_mutex_unlock(&ctx->lock);
+}
+
+int bt_testlib_att_write(struct bt_conn *conn, enum bt_att_chan_opt bearer, uint16_t handle,
+			 uint8_t *data, uint16_t size)
+{
+	int api_err;
+	struct bt_testlib_att_write_closure _ctx = {.params = {
+							    .handle = handle,
+							    .offset = 0,
+							    .func = att_write_cb,
+							    .data = data,
+							    .length = size,
+						    }};
```

Comment:
please don't. 
What's the point of automated tools (that already disagree w/ each other) if reviewers can still nit on formatting.
Also if we get an exception, anyone touching this line will have to get one too

Reply:
If checkpath is failing, then it's fine to keep as is (although we should investigate why it is failing, because 
```c
	struct bt_testlib_att_write_closure _ctx = {
		.params = {
			.handle = handle,
			.offset = 0,
			.func = att_write_cb,
			.data = data,
			.length = size,
		},
	};
```

Should be perfectly fine

Reply:
> What's the point of automated tools (that already disagree w/ each other)

The difference here is that checkpath is a official tool used by Zephyr's CI system, and clang-format is not officially supported (and probably won't be before it fixes a lot the issues it has).